### DISCUSSION TITLE
how does setting isolation level in transactions work in the background?

### DISCUSSION THREAD

**Discussion by steliosrammos** [prisma/prisma#9002] [Q&A] [answered]
Hey, I'm aware Prisma does not support setting the isolation level of transactions. As a current work-around I've seen the following being done: 

```ts
const SERIALIZE = 'SET TRANSACTION ISOLATION LEVEL SERIALIZABLE'
prisma.$transaction([
    models.$executeRaw(SERIALIZE), 
    models.$executeRaw(<some query>)
])
```

However, I was wondering how that would work for nested updates that, as far as I understand, also use transactions. I have tested something similar to the following and the query executes fine, however I don't know if the isolation level is set for the nested update or not. 

```ts
const SERIALIZE = 'SET TRANSACTION ISOLATION LEVEL SERIALIZABLE'
const updatedPost: Post = prisma.post.update({
  where: { id: 42 },
  data: {
    author: {
      connect: { email: 'alice@prisma.io' },
    },
  },
})

prisma.$transaction([
    models.$executeRaw(SERIALIZE), 
    updatedPost
])
```

I'd appreciate if anyone could shed some light on how this works in the background!

**Comment by steliosrammos** [comment]
Any way to bump this?

**Comment by garrensmith** [comment, marked as answer]
Hi @steliosrammos,

if you enable interactive transactions in your schema:

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["interactiveTransactions"]
}
```

The list of operations passed to a `$transaction` will be done in one interactive transaction. That includes bulk operations and nested operations.  

And if you use our latest release 3.10 any operations in a `$transaction` will always occur in one transaction even if you do not have interactive transactions enabled.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
